src/charts/legend.coffee

```diff
@@ -219,3 +228,10 @@ Ember.Charts.Legend = Ember.Mixin.create
       .text((d) -> d.label)
       .attr(@get 'legendLabelAttrs')
       .call(labelTrimmer.get 'trim')
+
+    # we use averageLegendLabelWidth in order to estimate how much to move the
+    # labels to make them seem roughly centered
+    totalLabelWidth = 0
+    legend.selectAll('text').each ->
+      totalLabelWidth += this.getComputedTextLength()
+    @set 'averageLegendLabelWidth', totalLabelWidth / @get 'legendItems.length'
```

Comment:
I moved the declaration of averageLegendLabelWidth further down to make it more clear but I think I need to have this in the redraw legend function so that it happens every time we redraw